src/common/Modal/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, memo, useCallback, useEffect } from 'react'
+import { createPortal } from 'react-dom'
+import styled from 'styled-components'
+
+type PortalModalPropsType = {
+  visible: boolean
+  setIsModalVisible: (value: boolean) => void
+  children: React.ReactNode
+}
+
+export const Modal: FC<PortalModalPropsType> = memo(({ visible, children, setIsModalVisible }) => {
+  const onClickHandler = useCallback(() => {
```

Comment:
set up prettier pls, because it's strange that there are no empty lines between functions and variables in component